docs/DeveloperGuide.md

```diff
@@ -239,6 +240,47 @@ _{more aspects and alternatives to be added}_
 _{Explain here how the data archiving feature will be implemented}_
 
 
+### Listing Events
+
+The List Events feature allows users to enter `listEvents` and update the UI display for events, showing a list of all 
+events sorted in their current order inside `ModelManager`. In the future, the command will take in a parameter,
+which specifies the field that the events list can be permanently sorted by.
+
+### Implementation
+The List Events feature is facilitated by `ListEventsCommand` which extends from Command. Additionally, it implements
+the following operation:
+
+* `ModelManager#updateFilteredEventList`  — Updates the predicate inside `ModelManager`'s filtered event list 
+to modify and sort which and how events are shown.
+
+This operation is exposed in the `Model` interface as the method `Model#updateFilteredEventList`.
+
+The following sequence diagram shows how the listEvents operation works.
+
+![ListEventsSequenceDiagram](images/ListEventsSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the activities related to events are
+considered and shown in this activity diagram.
+</div>
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for ListEventsCommand should
+end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of the diagram.
+</div>
+
+### Design Considerations:
+
+**Aspect: If listEvents should sort events permanently:
+* **Alternative 1 (current choice):** Sort all events permanently.
+  * Pros: Easy to implement.
+  * Cons: Whenever events are added, they are always added as the last event in the event list. Once sorted, the order 
+  cannot be returned to the order which events were added.
+* **Alternative 2:** Sort all events temporarily.
+  * Pros: Users can choose if they want to sort their events permanently or temporarily.
+  * Cons: Hard to implement as a new form of storage or memory has to be created to maintain the relative ordering of
+    events initially. Furthermore, a new sort command must be created to sort events permanently.
```

Comment:
Alternative 2 sounds a little confusing to me. It suggests that there are 2 ways to sort events: permanently or temporarily? I think to keep things simple for the reader, Alternative 2 should just explain the pros and cons of temporary sorting, ignoring the concept of permanent sorting since that was covered in Alternative 1.